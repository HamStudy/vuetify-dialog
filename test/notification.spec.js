import Vue from 'vue'
import Vuetify from 'vuetify'
import { sleep } from './utils'
import VuetifyDialogPlugin from '../src'
Vue.use(Vuetify)

describe('manager', () => {
  let manager

  beforeAll(() => {
    Vue.use(VuetifyDialogPlugin)
    manager = Vue.prototype.$dialog
  })

  test('Test success notification', async () => {
    const dlg = await manager.notify.success('Test confirmation', {
      timeout: 100,
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(100)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test error notification with top-left position', async () => {
    const dlg = await manager.notify.error('Test confirmation', {
      timeout: 100,
      position: 'top-left',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(100)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test warning notification with bottom-right position', async () => {
    const dlg = await manager.notify.warning('Test confirmation', {
      timeout: 100,
      position: 'bottom-right',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(100)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
