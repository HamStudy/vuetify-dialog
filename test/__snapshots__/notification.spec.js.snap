// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test error notification with top-left position 1`] = `
<div
  class="vuedl-notification left vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <!---->
   
  <!---->
</div>
`;

exports[`manager Test info notification 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-leave vuedl-notification-fade-leave-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <!---->
   
  <!---->
</div>
`;

exports[`manager Test notification component 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <!---->
   
  <!---->
</div>
`;

exports[`manager Test success notification 1`] = `
<div
  class="vuedl-notification right vuedl-notification-fade-enter vuedl-notification-fade-enter-active"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <!---->
   
  <!---->
</div>
`;

exports[`manager Test warning notification with bottom-right position 1`] = `
<div
  class="vuedl-notification right"
  role="alert"
  style="max-width: 500px; z-index: 2000; bottom: 10px;"
>
  <!---->
   
  <!---->
</div>
`;
